packages/core/src/tools/error.ts

```diff
@@ -110,3 +141,21 @@ export function createHandlingStack(): string {
 
   return formattedStack!
 }
+
+type FormatErrorCausesParams = { errorObject: ErrorWithCause; parentSource: ErrorSource }
+
+export function flattenErrorCauses({ errorObject, parentSource }: FormatErrorCausesParams): RawErrorCause[] {
+  let currentError = errorObject
+  const causes: RawErrorCause[] = []
+  while (currentError?.cause && currentError.cause && causes.length < 10) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏no?
```suggestion
  while (currentError?.cause && causes.length < 10) {
```